docs/maps/api-ui.md

```diff
@@ -379,28 +379,57 @@ If the user is in mobile definition, the modal will open in full screen:
 
 ```ts
 WA.ui.actionBar.addButton(descriptor: {
-    id: string, 
+    id: string,
     label: string,
+    type: 'button'|'action',
+    imageSrc: string ?? null,
+    toolTip: string ?? null,
     clickCallback: (buttonActionBar: AddButtonActionBar) => void
 }): void
 ```
 
 - id: the id of the button action bar defined.
 - label: the label to display in the button.
+- type: the type of button ('button' / 'action'). By default is 'button'.
+- imageSrc: image of button associated, This parameter is nullable.
+- toolTip: label displayed above the action button. This parameter is nullable.
 - clickCallback: function called when the user clicks on the button. The callback is passed a `AddButtonActionBar` instance in parameter.
 
 With `AddButtonActionBar` defined as:
 ```ts
+/**
+ * Ennum of button type
+ */
+enum ActionBarButtonType {
+    button = "button",
+    action = "action",
+}
```

Comment:
I've had issues with enums in the past (mostly on the Woka body parts). I'm starting to consider them as evil.

See https://stackoverflow.com/questions/40275832/typescript-has-unions-so-are-enums-redundant

So would you be OK to switch that to a simple union type?

```suggestion
/**
 * Possible values of button type
 */
type ActionBarButtonType = "button" | "action";
```

Reply:
Wait... scrach that, we can do better! See my comment below about discriminated unions.